micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/common/response/executor/RestExecutor.groovy

```diff
@@ -36,10 +39,17 @@ final class RestExecutor<T> {
         try {
             return exchangeInternal(params, httpMethod, responseType).get()
         } catch (ExecutionException e) {
-            throw e.cause
+            propagate(e.cause)
         }
     }
 
+    private void propagate(Throwable throwable) {
+        if (throwable instanceof TimeoutException) {
```

Comment:
Good question. We believe having a separate exception for timeouts (e.g. some system is slow) might be beneficial. But when you said it, I am not so convinced anymore.